+nosnoc/+solver/MpccSolver.m

```diff
@@ -426,5 +458,415 @@ function print_infeasibility(obj, results)
+        end
+
+        function [solution, improved_point, b_stat] = check_b_stationarity(obj, s_elastic)
+            import casadi.*
+            mpcc = obj.mpcc;
+            nlp = obj.nlp;
+            x0 = nlp.w.mpcc_w().res;
+            f = mpcc.f;
+            x = nlp.w.mpcc_w(); lbx = nlp.w.mpcc_w().lb; ubx = nlp.w.mpcc_w().ub;
+            g = nlp.g.mpcc_g(); lbg = nlp.g.mpcc_g().lb; ubg = nlp.g.mpcc_g().ub;
+            G = mpcc.G;
+            H = mpcc.H;
+            G_old = full(obj.G_fun(x0, nlp.p.mpcc_p().val));
+            H_old = full(obj.H_fun(x0, nlp.p.mpcc_p().val));
+
+            if obj.opts.homotopy_steering_strategy == HomotopySteeringStrategy.DIRECT
+                a_tol = 1*sqrt(obj.opts.comp_tol);
+            else
+                a_tol = 1*sqrt(s_elastic);
+            end
+            ind_00 = G_old<a_tol & H_old<a_tol;
+            n_biactive = sum(ind_00);
+            ind_0p = G_old<H_old & ~ind_00;
+            ind_p0 = H_old<G_old & ~ind_00;
+            y_lpcc = H_old<G_old;
+            
+            lift_G = SX.sym('lift_G', length(G));
+            lift_H = SX.sym('lift_H', length(H));
+            g_lift = vertcat(lift_G - G, lift_H - H);
+
+            x = vertcat(x, lift_G, lift_H);
+            lblift_G = zeros(size(lift_G));
+            ublift_G = inf*ones(size(lift_G));
+            lblift_H = zeros(size(lift_G));
+            ublift_H = inf*ones(size(lift_G));
+
+            %TODO (@anton) maybe take advantage of vdx here.
+            ind_mpcc = 1:length(lbx);
+            ind_G = (1:length(lift_G))+length(lbx);
+            ind_H = (1:length(lift_H))+length(lbx)+length(lift_G);
+
+            lbx = vertcat(lbx, lblift_G, lblift_H);
+            ubx = vertcat(ubx, ublift_G, ublift_H);
+            G_init = G_old;
+            %G_init(ind_00 | ind_0p) = 0;
+            H_init = H_old;
+            %H_init(ind_00 | ind_p0) = 0;
+            x0 = vertcat(x0, G_init, H_init);
+
+            g = vertcat(g,g_lift);
+            lbg = vertcat(lbg,zeros(size(g_lift)));
+            ubg = vertcat(ubg,zeros(size(g_lift)));
+            
+            p = mpcc.p;
+            p0 = nlp.p.mpcc_p().val;
+
+            solver_settings.max_iter = 20;
+            solver_settings.tol = 1e-6;
+            solver_settings.Delta_TR_init = 10;
+            solver_settings.Delta_TR_min = 1e-4;
+            solver_settings.verbose_solver = 1;
+            solver_settings.tighten_bounds_in_lpcc = false;
+            solver_settings.BigM = 1e2;
+
+            if ~n_biactive
+                solver_settings.fixed_y_lpcc = y_lpcc;
+            end
+            %% The problem
+            rnlp = struct('x', x, 'f', f, 'g', g, 'comp1', lift_G, 'comp2', lift_H, 'p', p);
+            solver_initalization = struct('x0', x0, 'lbx', lbx, 'ubx', ubx,'lbg', lbg, 'ubg', ubg, 'p', p0, 'y_lpcc', y_lpcc);
+            solution = b_stationarity_oracle(rnlp,solver_initalization,solver_settings);
+            cpu_time_filterSMPCC2 = toc;
```

Comment:
change name of the variable to more descriptive one

Reply:
Removed as this is vestigial.